types/src/move_resource.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    access_path::{AccessPath, Accesses},
+    account_config,
+    language_storage::{StructTag, TypeTag},
+};
+use move_core_types::identifier::{IdentStr, Identifier};
+
+pub trait MoveResource {
+    const MODULE_NAME: &'static str;
```

Comment:
if we can also generalize modules as a trait, we can have something like `trait MoveResource<M: ResourceModule>` and get rid of the `MODULE_NAME` field. This would also help making the code more ready for change, if we ever want to make changes to a module, we can just change the ResourceModule instead of individual MoveResources (esp for Account module). But I'll leave that decision to Move team :P